src/layouts/EditHomepage.jsx

```diff
@@ -241,6 +242,34 @@ export default class EditHomepage extends Component {
 
           sections[sectionIndex][sectionType][field] = value;
 
+          // Set special error message if hero button has text but hero url is empty
+          // This needs to be done separately because it relies on the state of another field
+          if (sectionIndex === 0 && sectionType === 'hero' && field === 'url' && !value && this.state.frontMatter.sections[0].hero.button) {
```

Comment:
We could consider also adding this check for when the button is changed - currently users are still allowed to save if the url is empty but there is button text (happens most frequently if the hero section is changed from a dropdown to highlights + buttons)

Reply:
Good point. Fixed with [904044a](https://github.com/isomerpages/isomercms-frontend/pull/230/commits/904044a605acf85a6109f481e3fd34b0ed465399) and extended to infobar and infopic sections in [9910ee4](https://github.com/isomerpages/isomercms-frontend/pull/230/commits/9910ee4548157caf6377783fbe5d80feefb60626)